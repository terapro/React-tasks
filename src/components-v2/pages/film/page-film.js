import React, {Fragment} from 'react';
import {BodySection} from "src/components-v2/body/section/body-section";
import {Header} from "src/components-v2/header/header";

export const PageFilm = () => (
  <Fragment>
    <BodySection bgTheme>
      <Header/>
    </BodySection>
    <BodySection bgLight>
      film page
    </BodySection>
  </Fragment>
);