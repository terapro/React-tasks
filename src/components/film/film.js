import React from 'react';
import PropTypes from 'prop-types';
import {FilmPoster} from 'src/components/film/film-poster/film-poster.js';
import {FilmInfo} from 'src/components/film/film-info/film-info.js';

export const Film = ({active, info}) => {
  if (active) {
    return (<div className="film">
      <FilmPoster posterPath={info['poster_path']}/>
      <FilmInfo info={info}/>
    </div>);
  } else {
    return null;
  }
};
Film.propTypes = {
  info: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
  }),
  active: PropTypes.bool
};

Film.defaulProps = {
  active: false,
  info: { }
};
