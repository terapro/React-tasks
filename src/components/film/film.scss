.film {
  display: flex;
  margin: $elementPadding 0;
  justify-content: space-between;
  align-items: center;
  background: rgba(0,0,0,0.5);
}