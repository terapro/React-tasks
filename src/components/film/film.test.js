import React from 'react';
import {Film} from 'src/components/film/film.js';
import {shallow} from 'enzyme';

const mockFilm = {
  "id": 333339,
  "title": "Ready Player One",
  "tagline": "A better reality awaits.",
  "vote_average": 8.1,
  "vote_count": 617,
  "release_date": "2018-03-28",
  "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
  "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
  "budget": 175000000,
  "revenue": 0,
  "genres": [
    "Adventure",
    "Science Fiction",
    "Action"
  ],
  "runtime": 140
};

describe('<Film />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<Film active info={mockFilm} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should not render the component without active prop', () => {
      expect(shallow(<Film info={mockFilm} />).is('.film')).toBe(false);
    });
  });
});