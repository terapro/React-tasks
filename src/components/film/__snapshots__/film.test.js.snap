// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Film /> Rendering should render the component 1`] = `
<div
  className="film"
>
  <FilmPoster
    posterPath="https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg"
  />
  <FilmInfo
    info={
      Object {
        "budget": 175000000,
        "genres": Array [
          "Adventure",
          "Science Fiction",
          "Action",
        ],
        "id": 333339,
        "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
        "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
        "release_date": "2018-03-28",
        "revenue": 0,
        "runtime": 140,
        "tagline": "A better reality awaits.",
        "title": "Ready Player One",
        "vote_average": 8.1,
        "vote_count": 617,
      }
    }
  />
</div>
`;
